pkg/dynamicinstrumentation/codegen/c/event.h

```diff
@@ -0,0 +1,19 @@
+#ifndef DI_EVENT_H
+#define DI_EVENT_H
+
+#include "ktypes.h"
+
+struct event {
+    struct base_event base;
+    char output[PARAM_BUFFER_SIZE];
+};
+
+struct expression_context {
+    __u64 *output_offset;
```

Comment:
why this needs to be a pointer?

Reply:
The output_offset is incremented in the location expression function bodies

Reply:
https://github.com/DataDog/datadog-agent/blob/grantseltzer/DEBUG-2982-multiple-dereference-support-2/pkg/dynamicinstrumentation/codegen/c/expressions.h#L43